sdk/java-sdk/src/main/scala/kalix/javasdk/impl/DiscoveryImpl.scala

```diff
@@ -235,13 +226,41 @@ class DiscoveryImpl(system: ActorSystem, services: Map[String, Service]) extends
     EntitySettings.SpecificSettings.ReplicatedEntity(ReplicatedEntitySettings(writeConsistency))
   }
 
-  private def loadDescriptorsWithSource(path: String): Map[String, DescriptorProtos.FileDescriptorProto] =
+  override def proxyTerminated(in: Empty): Future[Empty] = {
+    log.debug("Proxy terminated")
+    proxyTerminatedRef.get().trySuccess(Done)
+    Future.successful(Empty.defaultInstance)
+  }
+}
+
+object DiscoveryImpl {
+
+  private[impl] def fileDescriptorSet(services: Iterable[Service], userDescPath: String, log: Logger) = {
+    val descriptorsWithSource = loadDescriptorsWithSource(userDescPath, log)
+    val allDescriptors =
+      AnySupport.flattenDescriptors(services.flatMap(s => s.descriptor.getFile +: s.additionalDescriptors).toSeq)
```

Comment:
(here is the 'real' change, the rest is just to make `s.additionalDescriptors` available and make things easier to test - though I didn't succeed in actually creating a satisfying test for this yet)